client/clientd/src/main.rs

```diff
@@ -99,3 +102,19 @@ async fn wait_block_height(
     client.await_consensus_block_height(payload.height).await;
     Json(RpcResult::Success(json!("")))
 }
+
+async fn peg_in(
+    Extension(state): Extension<Arc<State>>,
+    payload: Json<PegInPayload>,
+) -> impl IntoResponse {
+    let client = &state.client;
+    let mut rng = state.rng.clone();
+    let txout_proof = payload.0.txout_proof;
+    let transaction = payload.0.transaction;
+    let txid = client
+        .peg_in(txout_proof, transaction, &mut rng)
+        .await
+        .unwrap(); //TODO: handle unwrap()
```

Comment:
Let's not unwrap here. If someone passed a bad pegin proof here it would crash clientd permanently.

I did a little prototype about how you can return `Result` from these handler functions: https://github.com/fedimint/fedimint/commit/daa19019e4973b803bd3043bb90c8ef60e67e004

I think it would be best to have some kind of middleware that can map `anyhow` rust errors to json-rpc error messages. Maybe a combination of `handle_anyhow_error` and "layers" described on the [Axum errorhandling docs page](https://docs.rs/axum/latest/axum/error_handling/index.html)

Reply:
In the final version I wanted to handle the error like [this](https://github.com/fedimint/fedimint/pull/183/files#diff-03e5ada38b52857ced3bfd584c0e3d9226844e0dbd362e9a6abe68c61318eec3R157-R174). I can't avoid doing unwraps if I want to do small PRs. So my general approach was to have a "success and failure" enum. Like this it is easier to deserialize the response later in the cli.

Reply:
My response returning is super weird though. I remember I only did that because I needed a uniform wrapper (`RpcResult`) to deserialize. I think I'll go with a `ClientdError` and `ClientdResult` which implement `IntoResponse`, like this I can use your suggestion too

Reply:
> i wanted to handle the error like this

You were also repeating code frequently for the actual error handling e.g. [here](https://github.com/fedimint/fedimint/pull/183/files#diff-03e5ada38b52857ced3bfd584c0e3d9226844e0dbd362e9a6abe68c61318eec3R196-R200). Ideally that could happen in once place.

> `ClientdError` and `ClientdResult` which implement `IntoResponse`

You only need `ClientdError`. Then return `Result<impl IntoResponse, ClientdError>`. I think that's good for now. In the future we can get smarter about how we present errors and consume them from the clientd client. For now, let's just not panic.